clang/lib/InstallAPI/Frontend.cpp

```diff
@@ -16,6 +16,74 @@ using namespace llvm::MachO;
 
 namespace clang::installapi {
 
+GlobalRecord *FrontendRecordsSlice::addGlobal(
+    StringRef Name, RecordLinkage Linkage, GlobalRecord::Kind GV,
+    const clang::AvailabilityInfo Avail, const Decl *D, const HeaderType Access,
+    SymbolFlags Flags) {
+
+  auto *GR = llvm::MachO::RecordsSlice::addGlobal(Name, Linkage, GV, Flags);
+  if (!FrontendRecords.contains(GR))
```

Comment:
You could optimize this a bit by always inserting. `insert` doesn't change anything if the key already exists.

Reply:
What if the record already exists? Should the attributes match or be merged?

Reply:
The way tapi handles this today is that none of the `frontend` attributes are updated or even checked if they differ. Since this is not a problem area, I figured to keep that behavior.